src/js/stores/DialogStore.js

```diff
@@ -0,0 +1,104 @@
+var EventEmitter = require("events").EventEmitter;
+var lazy = require("lazy.js");
+
+var AppDispatcher = require("../AppDispatcher");
+var DialogEvents = require("../events/DialogEvents");
+
+function addDialog(dialogs, message, dialogId) {
+  dialogs.push({id: dialogId, message: message});
+}
+
+function removeDialog(dialogs, dialogId) {
+  dialogs = dialogs.filter(function (dialog) {
+    return dialog.id !== dialogId;
+  });
+}
+
+var DialogStore = lazy(EventEmitter.prototype).extend({
+  dialogs: [],
+  handleUserResponse: function (dialogId, acceptCallback, dismissCallback) {
+    dismissCallback = dismissCallback || function () {};
+
+    function onAccept(id, value = null) {
+      if (dialogId === id) {
+        acceptCallback(value);
+        removeListeners();
+      }
+    }
+
+    function onDismiss(id) {
+      if (dialogId === id) {
+        dismissCallback();
+        removeListeners();
+      }
+    }
+
+    function removeListeners() {
+      DialogStore.removeListener(DialogEvents.CONFIRM_ACCEPT, onAccept);
+      DialogStore.removeListener(DialogEvents.CONFIRM_CLOSE, onDismiss);
+      DialogStore.removeListener(DialogEvents.PROMPT_ACCEPT, onAccept);
+      DialogStore.removeListener(DialogEvents.PROMPT_CLOSE, onDismiss);
+    }
+
+    DialogStore.on(DialogEvents.CONFIRM_ACCEPT, onAccept);
+    DialogStore.on(DialogEvents.CONFIRM_CLOSE, onDismiss);
+    DialogStore.on(DialogEvents.PROMPT_ACCEPT, onAccept);
+    DialogStore.on(DialogEvents.PROMPT_CLOSE, onDismiss);
+  }
+}).value();
+
+AppDispatcher.register(function (action) {
+  switch (action.actionType) {
+    case DialogEvents.ALERT:
+      addDialog(DialogStore.dialogs, action.message, action.dialogId);
```

Comment:
I'm a bit slow today so bear with me :stuck_out_tongue: 
Can you help me understand why you prefer to have `addDialog` & `removeDialog` outside of `DialogStore` and have them mutate the `DialogStore.dialogs` array directly? 

Basically, what are the downsides of opting for a `DialogStore.addDialog(action.message, action.dialogId)` here? Is it to avoid exposing it?